sky/provision/kubernetes/instance.py

```diff
@@ -0,0 +1,713 @@
+
+    for pod_name, pod in pods.items():
+        logger.debug(f'Terminating instance {pod_name}: {pod}')
+        if _is_head(pod) and worker_only:
+            continue
+        _terminate_node(namespace, pod_name)
+
+
+def get_cluster_info(
+        region: str,
+        cluster_name_on_cloud: str,
+        provider_config: Optional[Dict[str, Any]] = None) -> common.ClusterInfo:
+    del region  # unused
+    assert provider_config is not None
+    namespace = _get_namespace(provider_config)
+    tag_filters = {
+        TAG_RAY_CLUSTER_NAME: cluster_name_on_cloud,
+    }
+
+    running_pods = _filter_pods(namespace, tag_filters, ['Running'])
+    pods: Dict[str, List[common.InstanceInfo]] = {}
+    head_pod_name = None
+
+    port_forward_mode = kubernetes_enums.KubernetesNetworkingMode.PORTFORWARD
+    network_mode_str = skypilot_config.get_nested(('kubernetes', 'networking'),
+                                                  port_forward_mode.value)
+    network_mode = kubernetes_enums.KubernetesNetworkingMode.from_str(
+        network_mode_str)
+    external_ip = kubernetes_utils.get_external_ip(network_mode)
+    port = 22
+    if not provider_config.get('use_internal_ips', False):
+        port = kubernetes_utils.get_head_ssh_port(cluster_name_on_cloud,
+                                                  namespace)
+
+    head_pod_name = None
+    cpu_request = None
+    for pod_name, pod in running_pods.items():
+        internal_ip = pod.status.pod_ip
+        pods[pod_name] = [
+            common.InstanceInfo(
+                instance_id=pod_name,
+                internal_ip=internal_ip,
+                external_ip=(None if network_mode == port_forward_mode else
+                             external_ip),
+                ssh_port=port,
+                tags=pod.metadata.labels,
+            )
+        ]
+        if pod.metadata.labels[TAG_RAY_NODE_KIND] == 'head':
+            head_pod_name = pod_name
+            head_spec = pod.spec
+            assert head_spec is not None, pod
+            cpu_request = head_spec.containers[0].resources.requests['cpu']
+
+    assert cpu_request is not None, 'cpu_request should not be None'
+
+    ssh_user = 'sky'
+    get_k8s_ssh_user_cmd = ['/bin/sh', '-c', ('echo $(whoami)')]
+    assert head_pod_name is not None
+    ssh_user = run_command_on_pods(head_pod_name, namespace,
+                                   get_k8s_ssh_user_cmd)
+    ssh_user = ssh_user.strip()
+    logger.debug(
+        f'Using ssh user {ssh_user} for cluster {cluster_name_on_cloud}')
+
+    return common.ClusterInfo(
+        instances=pods,
+        head_instance_id=head_pod_name,
+        ssh_user=ssh_user,
+        # We manually set object-store-memory=500000000 to avoid ray from
+        # allocating a very large object store in each pod that may cause
+        # problems for other pods.
+        custom_ray_options={
+            'object-store-memory': 500000000,
+            'num-cpus': cpu_request,
```

Comment:
I'm not fully familiar with the new provisioner code path, but how are these custom_ray_options used? I'm wondering if we also need to add `num-gpus: gpu_request` here...

Reply:
The gpu related configs are set in the instance_setup for the ray cluster, as those are also required for other clouds, but these options seem only be required by ray cluster on kubernetes.
https://github.com/skypilot-org/skypilot/blob/0df27493538cff7da5571edd68ee66e74a4e0e9b/sky/provision/instance_setup.py#L215-L226

Note that the GPUs are configured by `--resources=\'{custom_resource}\'`